### PR TITLE
post-comment-by-article-id

### PR DESCRIPTION
**Pull request by LW95x**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -151,3 +151,51 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+
+    return request(app)
+      .post("/api/articles/not_an_id/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad request");
+      });
+  });
+  test("POST 404: not found, article_id does not exist in the table", () => {
```
Comment (nc-poppy-m) [inline __tests__/integration.test.js:187]: Good 404 - another 404 to consider is if the username doesn't exist

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -151,3 +151,51 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body.author).toBe("butter_bridge");
+        expect(response.body.body).toBe("this is a test comment");
+        expect(response.body.article_id).toBe(1);
+        expect(response.body.votes).toBe(0);
+      });
+  });
+  test("POST 400: invalid article_id input", () => {
```
Comment (nc-poppy-m) [inline __tests__/integration.test.js:173]: This is more of a 'missing property' 400. Could you try implementing a missing property (ideally not `article_id`) 400 as well as an invalid `article_id` 400?

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -151,3 +151,51 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("POST 201: should add a comment to an existing article", () => {
```
Comment (nc-poppy-m) [inline __tests__/integration.test.js:156]: What would happen if I added unnecessary properties?
``` 
const newComment = {
      username: "butter_bridge",
      body: "this is a test comment",
      extra_property: "banana",
    }

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Good work so far, nearly there!
- I've added some feedback about testing, including 3-4 new tests. Please test for these behaviours before resubmitting this ticket (consider if the errors should be handled with your PSQL error handler or your custom error handler).
- I can't see your updated endpoints.json - make sure it's included when you resubmit

File: __tests__/integration.test.js [hunk 4]
```diff
@@ -198,4 +198,35 @@ describe("POST /api/articles/:article_id/comments", () => {
       expect(response.body.msg).toBe("that ID does not exist");
     });
   })
+  test("POST 404: username does not exist", () => {
```
Comment (nc-mick-fay) [inline __tests__/integration.test.js]: This test description says 404, but you're testing for 400

**Review by nc-mick-fay — CHANGES_REQUESTED** [review]
Hi Liam, we're just missing one test:

- [ ] Status 400 - incomplete body: What happens if you send a request that doesn't have either a username or a body?

Besides that, all your testing is really solid. I really like your chaining of the checkArticleExists model to distinguish between error messages

**Review by nc-poppy-m — APPROVED** [review]
Superb, merge when ready!

**Pull request merged**